pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -677,7 +677,16 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
 					return err
 				}
 				var preRestoreObjects []runtime.Unstructured
-				if !nfs {
+				if nfs {
+					if restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyRetain {
+						// Skip pv/pvc if replacepolicy is set to retain to avoid creating with empty object
+						backupVolInfos, existingRestoreVolInfos, err = a.skipVolumesFromRestoreList(restore, preRestoreObjects, driver, vInfos)
```

Comment:
let us send the object variable instead of preRestoreObjects even both are nil, to avoid confusion.